driver-core/src/main/java/com/datastax/driver/core/GuavaCompatibility.java

```diff
@@ -0,0 +1,215 @@
+                Attributes mainAttributes = manifest.getMainAttributes();
+                String symbolicName = mainAttributes.getValue("Bundle-SymbolicName");
+                if ("com.google.guava".equals(symbolicName)) {
+                    return manifest;
+                }
+            }
+            throw new DriverInternalError("Error while looking up Guava manifest: " +
+                    "no manifest with symbolic name 'com.google.guava' found in classpath.");
+        } catch (Exception e) {
+            throw new DriverInternalError("Error while looking up Guava manifest", e);
+        } finally {
+            if (in != null) {
+                try {
+                    in.close();
+                } catch (IOException e) {
+                    // ignore
+                }
+            }
+        }
+    }
+
+    private static String getBundleVersion(Manifest manifest) {
+        return manifest.getMainAttributes().getValue("Bundle-Version");
+    }
+
+    private static String stripQualifiers(String fullVersion) {
+        Matcher matcher = GA_VERSION_EXTRACTOR.matcher(fullVersion);
+        if (matcher.matches()) {
+            return matcher.group(1);
+        } else {
+            throw new DriverInternalError(String.format("Could not strip qualifiers from full Guava version %s", fullVersion));
+        }
+    }
+
+    private static class Version18OrLower extends GuavaCompatibility {
+
+        @Override
+        public <V> ListenableFuture<V> withFallback(ListenableFuture<? extends V> input,
+                                                    final AsyncFunction<Throwable, V> fallback) {
+            return Futures.withFallback(input, new FutureFallback<V>() {
+                @Override
+                public ListenableFuture<V> create(Throwable t) throws Exception {
+                    return fallback.apply(t);
+                }
+            });
+        }
+
+        @Override
+        public <V> ListenableFuture<V> withFallback(ListenableFuture<? extends V> input,
+                                                    final AsyncFunction<Throwable, V> fallback,
+                                                    Executor executor) {
+            return Futures.withFallback(input, new FutureFallback<V>() {
+                @Override
+                public ListenableFuture<V> create(Throwable t) throws Exception {
+                    return fallback.apply(t);
+                }
+            }, executor);
+        }
+
+        @Override
+        public <I, O> ListenableFuture<O> transformAsync(ListenableFuture<I> input, AsyncFunction<? super I, ? extends O> function) {
+            return Futures.transform(input, function);
+        }
+
+        @Override
+        public <I, O> ListenableFuture<O> transformAsync(ListenableFuture<I> input, AsyncFunction<? super I, ? extends O> function, Executor executor) {
+            return Futures.transform(input, function, executor);
```

Comment:
Found one occurrence of this method in `com.datastax.driver.core.AsyncQueryTest#connectAndQuery`